File: programmers/lv1/옹알이(2).js
// https://school.programmers.co.kr/learn/courses/30/lessons/133499

function solution(babbling) {
  let answer = 0;
  const possible = ["aya", "ye", "woo", "ma"];
  const notPossible = ["ayaaya", "yeye", "woowoo", "mama"];
  for (let i = 0; i < babbling.length; i++) {
    for (let j = 0; j < possible.length; j++) {
      if (!babbling[i].includes(notPossible[j]))
        babbling[i] = babbling[i].replaceAll(possible[j], " ");
    }
    if (babbling[i].trim() === "") answer++;
  }
  return answer;
}

File: programmers/lv1/옹알이(2).py
def solution(babbling):
    answer = 0
    o = ['aya', 'ye', 'woo', 'ma']
    con = ['ayaaya', 'yeye', 'woowoo', 'mama']
    for b in babbling:
        for i,j in zip(o, con):
            if j not in b:
                b = b.replace(i, '*')
        if b.count('*') == len(b):
            answer += 1
    return answer